### PR TITLE
Fix mem leak in HandleTlsResumption

### PR DESCRIPTION
**Pull request by embhorn**
This fixes an issue from zd13139

### CODE REVIEW

**Comment by embhorn** [comment]
Please hold off on merge until customer confirms fix

**Review by SparkiDev — APPROVED** [review]

**Comment by SparkiDev** [comment]
I took a look at the code and agree this is all that is needed.

File: src/internal.c [hunk 1]
```diff
@@ -28958,13 +28968,20 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
...
+                    wolfSSL_SESSION_free(session);
+                }
             #endif
                 return UNSUPPORTED_SUITE;
             }
         #endif
 
         #ifdef HAVE_EXT_CACHE
-            wolfSSL_SESSION_free(session);
+            if (gotSess) {
```
Comment (SparkiDev) [inline src/internal.c:28982]: Session could be freed at end of if statement.
The last else branch can free after loop and set session to NULL.

**Review by SparkiDev — DISMISSED** [review]

File: src/internal.c [hunk 2]
```diff
@@ -28885,25 +28885,31 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
...
-        session = GetSession(ssl, ssl->arrays->masterSecret, 1);
-        #ifdef HAVE_SESSION_TICKET
-            if (ssl->options.useTicket == 1) {
-                session = &ssl->session;
-            } else if (bogusID == 1 && ssl->options.rejectTicket == 0) {
-                WOLFSSL_MSG("Bogus session ID without session ticket");
-                return BUFFER_ERROR;
-            }
+    #ifdef HAVE_EXT_CACHE
+        byte gotSess = 0;
```
Comment (dgarske) [inline src/internal.c:28897]: Some compilers will not like the `(void)bogusID` above variable declarations.

**Review by dgarske — CHANGES_REQUESTED** [review]
Otherwise looks great.

File: src/internal.c (hunk 2 above)
Comment (embhorn) [reply src/internal.c:28897]: Check

**Comment by embhorn** [comment]
Ready to merge

**Review by dgarske — APPROVED** [review]

**Pull request merged**